pangeo_forge/recipe.py

```diff
@@ -156,6 +160,8 @@ class NetCDFtoZarrSequentialRecipe(BaseRecipe):
     xarray_concat_kwargs: dict = field(default_factory=dict)
     delete_input_encoding: bool = True
     fsspec_open_kwargs: dict = field(default_factory=dict)
+    process_input: Optional[callable] = None
```

Comment:
```suggestion
    process_input: Optional[Callable[[xr.Dataset, str], xr.Dataset] = None
```

(just guessing on the type. But I think that's the syntax for typing the inputs and outputs of the callable.)